milatools/cli/commands.py

```diff
@@ -81,478 +95,778 @@ def main():
+    """Start an MLFlow server.
+
+    Arguments:
+        logdir: Path to the experiment logs
+    """
+
+    _standard_server(
+        logdir,
+        program="mlflow",
+        installers={
+            "pip": "pip install mlflow",
+        },
+        command="mlflow ui --backend-store-uri {path} --host {host} --port 0",
+        port_pattern="Listening at: http://[^:]+:([0-9]+)",
+        **kwargs,
+    )
+
+
+def aim(logdir: str, **kwargs: Unpack[StandardServerArgs]):
+    """Start an AIM server.
+
+    Arguments:
+        logdir: Path to the experiment logs
+    """
+    _standard_server(
+        logdir,
+        program="aim",
+        installers={
+            "pip": "pip install aim",
+        },
+        command="aim up --repo {path} --host {host} --port 0",
+        port_pattern=r"Open http://[^:]+:([0-9]+)",
+        **kwargs,
+    )
+
+
+def _get_server_info(
+    remote: Remote, identifier: str, hide: bool = False
+) -> dict[str, str]:
     text = remote.get_output(f"cat .milatools/control/{identifier}", hide=hide)
     info = dict(line.split(" = ") for line in text.split("\n") if line)
     return info
 
 
-@tooled
-def _get_server_info_command(remote):
-    # Server identifier output by the original mila serve command
-    # [positional]
-    identifier: Option
+class SortingHelpFormatter(argparse.HelpFormatter):
+    """Taken and adapted from https://stackoverflow.com/a/12269143/6388696"""
+
+    def add_arguments(self, actions):
+        actions = sorted(actions, key=operator.attrgetter("option_strings"))
+        # put help actions first.
+        actions = sorted(
+            actions, key=lambda action: not isinstance(action, _HelpAction)
+        )
+        super().add_arguments(actions)
+
 
-    return identifier, _get_server_info(remote, identifier)
+def _add_standard_server_args(parser: ArgumentParser):
+    parser.add_argument(
+        "--alloc",
+        nargs=argparse.REMAINDER,
+        help="Extra options to pass to slurm",
+        metavar="VALUE",
+    )
+    parser.add_argument(
+        "--job",
+        type=str,
+        default=None,
+        help="Job ID to connect to",
+        metavar="VALUE",
+    )
+    parser.add_argument(
+        "--name",
+        default=None,
+        type=str,
+        help="Name of the persistent server",
+        metavar="VALUE",
+    )
+    parser.add_argument(
+        "--node",
+        type=str,
+        default=None,
+        help="Node to connect to",
+        metavar="VALUE",
+    )
+    parser.add_argument(
+        "--persist",
+        action="store_true",
+        help="Whether the server should persist or not",
+    )
+    parser.add_argument(
+        "--port",
+        type=int,
+        default=None,
+        help="Port to open on the local machine",
+        metavar="VALUE",
+    )
+    parser.add_argument(
+        "--profile",
+        default=None,
+        type=str,
+        help="Name of the profile to use",
+        metavar="VALUE",
+    )
```

Comment:
You also need to add the arguments for `forward`, because `standard_server` calls `forward` at the end. These arguments should then be explicitly added to the call to `forward` (coleo didn't need to do that because it collects arguments recursively in the call tree, so any function that calls forward automatically gains `--port`).

Reply:
How can I be still missing an argument if the help strings are the same as before?